.github/workflows/publish-commit-bottles.yml

```diff
@@ -335,11 +336,23 @@ jobs:
 
           echo "head_sha=$(git rev-parse HEAD)" >> "$GITHUB_OUTPUT"
 
-      - name: generate build provenance
-        uses: github-early-access/generate-build-provenance@main
+      - name: Generate build provenance
+        # GitHub Attestations are still in early public access, and we've seen
+        # some sporadic errors when generating attestations.
+        # Rather than fail outright, we allow this step to fail and fall
+        # through to a last-ditch retry below. Longer term, we should put this
+        # above the 'pr-pull' step, which should be split into separate 'pull'
+        # and 'push to GitHub Packages' phases.
+        continue-on-error: true
```

Comment:
I think this is the right way to continue if this step fails. The other approach is to have the last ditch step use `if: failure()`, which AFAIK means it'll run if _any_ step fails, not just the last one.

Reply:
Yeah, agreed. I think the team working on this at GH has been tackling it though internal retries, so I suspect it'll get better over time.

Reply:
there is no side effect of ignoring the action step failure?

Reply:
I don't believe so -- if it fails it should fall through to the "last ditch" version, which will then hard-fail the action if it also fails. But maybe I missed something?